docs/DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,258 @@
+
+openshift-virtualization-tests is a public repository under the [RedHatQE organization](https://github.com/RedHatQE) on GitHub.
+
+The project is structured as follows:
+- [tests](../tests): Base directory for pytest tests
+  - Each component has its own directory
+  - Each feature has its own directory
+- [utilities](../utilities): Base directory for utility functions
+  - Each module contains a set of utility functions related to a specific topic, for example:
+    - [infra](../utilities/infra.py): Infrastructure-related (cluster resources) utility functions
+    - [constants](../utilities/constants.py): Constants used in the project
+- [docs](../docs): Documentation
+- [py_config](../tests/global_config.py) contains tests-specific configuration which can be controlled from the command line.
+Please refer to [pytest-testconfig](https://github.com/wojole/pytest-testconfig) for more information.
+
+
+# Contribution
+To contribute code to the project:
+
+## Pull requests
+- Fork the project and work on your forked repository
+- Before submitting a new pull request:
+  - Make sure you follow the [Style guide](STYLE_GUIDE.md)
+  - Make sure you have [pre-commit](https://pre-commit.com/) package installed
+  - Make sure you have [tox](https://tox.readthedocs.io/en/latest/) package installed
+- PRs that are not ready for review (but needed to be pushed for any reason) should have [WIP] in the title and labeled as "wip".
+  - When a PR is ready for review, remove the [WIP] from the title and remove the "wip" label.
+- PRs should be relatively small; if needed, the PRs should be split and depend on each other.
+  - Small PRs will get quicker review.
+  - Small PRs comments will be fixed quicker and merged quicker.
+  - Both the reviewer and the committer will benefit from this.
+- When a refactor is needed as part of the current PR, the refactor should be done in another PR and the current PR should be rebased on it.
+- Please address each comment in code review
+  - If a comment was addressed and accepted, please comment as done and resolve.
+  - If a comment was addressed and rejected or additional discussion is needed, add your input and do not resolve the comment.
+  - To minimize the number of comments, please try to address all comments in one PR.
+- Before a PR can be merged:
+  - PRs must be verified and marked with "verified" label.
+  - PRs must be reviewed by at least two reviewers other than the committer.
+  - All CI checks must pass.
+
+## Branching strategy
+The project follows Red Hat Openshift Virtualization versions lifecycle.
+If needed, once your PR is merged to `main`, cherry-pick your PR to the relevant branch(es).
+
+
+## Python
+- Reduce duplicate code, before writing new function search for it, probably someone already wrote it or one that should serve your needs.
+  - The project uses external packages that may already have a functionality that does what you need.
+- When using a variable more than once, save it and reuse.
```

Comment:
This looks too strange and may confuse coders in thinking they work on a starship to mars.
"more than once" is too generic and is badly interpreted by many contributors.
Here some alternatives:
- Use meaningful names for operations (functions), data (variables) and objects (e.g. class) so they will express intent and read well.
- When appropriate, reduce system load by caching the data. E.g. access API, calculate complex CPU or IO bound data.
  - Avoid optimizing for CPU access to data (e.g. doing `myattribute = foo.myattribute` just not save a deference.